### PR TITLE
z-input: add validity state in inputChange and stopTyping events

### PR DESCRIPTION
**Pull request by mikelina**

### CODE REVIEW

File: src/components/inputs/z-input/index.tsx [hunk 1]
```diff
@@ -125,13 +125,13 @@ export class ZInput {
     if (!this.isTyping) {
       this.emitStartTyping();
     }
-
+    let validity = this.hostElement.shadowRoot.querySelector("input").validity;
```
Comment (edoardocavazza) [inline src/components/inputs/z-input/index.tsx:128]: Not sure the `'input'` selector is enough to query all possible input elements. We should add `textarea`, `select` and `[contenteditable]`

File: src/components/inputs/z-input/index.tsx [hunk 2]
```diff
@@ -144,9 +144,12 @@ export class ZInput {
 
   /** Emitted when user stops typing, returns value */
```
Comment (vpasquino) [inline src/components/inputs/z-input/index.tsx:145]: Update return values (value, validity)

Comment (mikelina) [reply src/components/inputs/z-input/index.tsx]: Updated

**Comment by vpasquino** [comment]
> Approved with condition: we should open an issue to tackle the compatibility of the `required` attribute with the custom select component.

@edoardocavazza agreed! I started a little refactor of the select input (soon we're going to have an autocomplete select, and I think it's going to become a complex component), I believe we can integrate this task in the refactoring.

**Pull request merged**